### DISCUSSION TITLE
connections/v7/admin/secure/t_azure_oidc_websphere

### DISCUSSION THREAD

**Discussion by giscus** [HCL-TECH-SOFTWARE/HCLSoftware-Documentation-Home#25] [Q&A]
# connections/v7/admin/secure/t_azure_oidc_websphere

Single sign-on is accomplished by setting up a trust relationship between the Connections server and Microsoft Azure using the WebSphere OpenID Connect Relying Party Trust Association Interceptor (OIDC Relying Party TAI). This requires that the WebSphereOIDCRP application is installed on each cluster.

https://help.hcltechsw.com/connections/v7/admin/secure/t_azure_oidc_websphere.html

**Comment by barbj** [comment]
See the latest version of the OIDC config steps here: https://www.ibm.com/docs/en/was-nd/9.0.5?topic=users-configuring-openid-connect-relying-party

1. giving the ".ear" extension to each app in step 1 is misleading.  It makes it look like there are 4 EARs installed instead of one EAR installed 4 times.  You should remove the extension. I made the mistake of telling a customer that those EARs are not owned by WebSphere, but by HCL.  You do not need that coming back on you, do you?
2. to reduce trouble reports to both HCL and IBM, instruct the customer to install the latest TAI: https://www.ibm.com/support/pages/node/290565 .  That is now the very first step in the latest config steps.
3. don't recommend setting the signatureValue property , or else set it to HEADER
4. In step 4, did you intend to say to set the class on com.ibm.websphere.security.DeferTAItoSSO, or on com.ibm.websphere.security.InvokeTAIbeforeSSO?  If you set the class on DeferTAItoSSO, it has no effect unless the class is set on both properties.  See https://www.ibm.com/docs/en/was/9.0.5?topic=realm-security-custom-properties#com.ibm.websphere.security.DeferTAItoSSO .  Currently, only SAML is enabled to recognize both before and after SSO.  So, first decide what you want to do: invoke 
before or after LTPA.  If before, set the class on InvokeTAIbeforeSSO.  If after, don't set anything. 
5. Why are you telling them to set all the properties individually instead of using discovery?  Setting properties individually is very error-prone, especially since they have to replace the tenant in all those property values.  The Azure/Entra ID discovery endpoint is: https://login.microsoftonline.com/{tenant}/v2.0/.well-known/openid-configuration .  If you look at the def for discoveryEndpointUrl on https://www.ibm.com/docs/en/was-nd/9.0.5?topic=SSAW57_9.0.5/com.ibm.websphere.nd.multiplatform.doc/ae/csec_oidprop.html , you'll see that there are 10 properties that you do not have to set if you use discovery.  If you do this, I suggest adding: revokeEndpointEnabled=false and userinfoEndpointEnabled=false
6. In the value for interceptedPathFilter, you need to concatenate the lines or else make a note that they have to concatenate them when setting the value.
7. Ditto for excludedPathFilter
8. You're setting both jwkEndpointUrl and signVerifyAlias.  Why?
9. provider_1.scope says to do a string replacement as part of the api value, but provider_2.scope and provider_3.scope do not.  Is that intended?
9. have them set provider_1, 2, and 3.useRealm=WAS_DEFAULT, then remove step 12 and 14
10. In step 13g, you say to configure the alias for which the SSL cert for the token endpoint is stored be configured on provider_1.signVerifyAlias.  Those two might happen to be the same, but the certificate to verify the JWT that is issued by an Azure tenant is not necessarily the same as the SSL certificate for the token endpoint.  I think that you should remove step 13g and remove recommending setting signVerifyAlias for each provider in the table.  The JWT verification cert comes from the jwk that comes either from discovery or is configured directly in the props.

**Comment by cg39652** [comment]
For the scope, why "openid profile email api://{client_id}/default" ? Setting this in WebSphere causes 404 errors as that context path is not defined in the WebSphere plugin. Customer got it working by setting scope to "openid profile email https://{connections-hostname}" so is the URI arbitrary as long as it matches between Azure and WebSphere?

Also, will this work for non-SSO domain users? SPNEGO has the NO_SPNEGO redirect for non SSO users so what is the mechanism for OIDC?
